### ISSUE TITLE
Make warning for required fields not being set more intuitive (inline with page)

### ISSUE THREAD

**Issue by mrueschman** [remomueller/slice#212]
Right now if I try to submit a form with a required field left blank you get this type of warning:

![image](https://cloud.githubusercontent.com/assets/707167/8190438/725a84d8-1430-11e5-902a-feffa5ecab6b.png)

**Comment by remomueller** [comment]
So better inline notification for sheet entry?

**Comment by mrueschman** [comment]
Guess I missed the comment, but I came back to GitHub today to re-make this issue (good thing I checked archives).

Yes, better inline notifications! I say leave these popup warnings for the really serious stuff, like deleting sheets/subjects/etc and the variable changes that affect data.

**Comment by remomueller** [comment]
Hi @mrueschman, moving this issue to the next release, as it'll require more of a UI redesign to better handle notifications. As for the popup messages for deleting items, I was thinking of perhaps removing the extra popup, and then providing an "Undo" link like we've started doing in other places instead. (Could be a separate issue).

**Comment by remomueller** [comment]
The following fields still need to be integrated with the new backend validation:

- [x] Date
- [x] Integer
- [x] Numeric
- [x] Signature
- [ ] Calculated
- [ ] Checkbox
- [ ] Dropdown
- [ ] File
- [ ] Grid
- [ ] Radio
- [ ] String
- [ ] Text
- [ ] Time of Day
- [ ] Time Duration

**Comment by remomueller** [comment]
From @mrueschman: New sheets shouldn't start with required and recommended fields as being required (only once you skip past them).

**Comment by remomueller** [comment]
- [ ] New sheets shouldn't immediately show warning or danger highlights for blank variables
- [ ] Clicking on a checkbox should update the validation immediately
- [ ] Clicking on a radio buttons should update the validation immediately
- [ ] Drawing in a signature box should update the validation immediately after end of draw
